pkg/integrations/manager.go

```diff
@@ -280,6 +280,11 @@ func (m *Manager) ApplyConfig(cfg ManagerConfig) error {
 			shouldCollect = *common.ScrapeIntegration
 		}
 
+		// If Enabled is false then we should override the should collect
+		if p.cfg.CommonConfig().Enabled == false {
```

Comment:
Hmm, I don't think this is being done in the right place. The loop in 210 would still spawn the integration, which it probably shouldn't be doing.

One way to handle this might be to add a function to return the list of only enabled integrations and iterate over that instead when calling `ApplyConfig`. That way disabling an integration and removing it from the file are treated as the same.

Reply:
My thought on that was looking at the documentation is that `enabled` flag enables the scraping but if you include the integration in the configuration with `enabled = false` it still will get the statistics but won't send them. I guess *should* there be a difference in disabling an integration vs removing it, if the answer is no then I will put in my original fix. Specifically with the agent integration having it available but not sending for debugging purposes seemed useful. Can def go either way though.

Reply:
Nah, the intent is that `enabled` drives whether the integration should be ran (so disabling it is the same as removing it), and `scrape_integration` determines if that running integration should be scraped by the Agent automatically.